core/application/impl/app_configuration_impl.cpp

```diff
@@ -1482,7 +1472,7 @@ namespace kagome::application {
         }
       }
 
-      if (find_argument(vm, "enable-thorough-pruning")) {
+      if (vm.count("enable-thorough-pruning")) {
```

Comment:
Can this be **regression**?
There was a problem, that `defaulted` must be checked, not `count`?